fenix/app/src/main/java/org/mozilla/fenix/compose/button/FloatingActionButton.kt

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.button
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.FloatingActionButton
+import androidx.compose.material.FloatingActionButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Floating action button, to be used inside a [Box], to allow aligning the FAB to the bottom end corner.
+ *
+ * @param modifier [Modifier] to be applied to the action button.
+ * @param icon [Painter] icon to be displayed inside the action button.
+ * @param label Text to be displayed next to the icon.
+ * @param onClick Invoked when the button is clicked.
+ */
+@Composable
+fun FloatingActionButton(
+    modifier: Modifier = Modifier,
+    icon: Painter,
+    label: String? = null,
+    onClick: () -> Unit,
+) {
+    FloatingActionButton(
+        onClick = onClick,
+        modifier = Modifier.then(modifier),
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        contentColor = FirefoxTheme.colors.textActionPrimary,
+        elevation = FloatingActionButtonDefaults.elevation(99.dp),
```

Comment:
where is this elevation of 99 derived from?

Reply:
Initially I added it to match the xml FAB [elevation](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/res/layout/component_tabstray_fab.xml#17) but it seems to work well with the default FAB elevation, so I removed it.